From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Wed, 25 Oct 2023 11:48:21 -0400
Subject: chore: expose UI to allow Electron to set dock side

This patch allows Electron to set the dock side via `webContents.openDevTools({ mode })`.

It also allows us to test certain aspects of devtools extensions. We should look for a way
to handle this without patching, but this is fairly clean for now and no longer requires
patching legacy devtools code.

diff --git a/front_end/entrypoints/main/MainImpl.ts b/front_end/entrypoints/main/MainImpl.ts
index 4e78b827080c081607f1fdcda7ed3396b0e23bc6..6c077e9d62dcc235e6642d5ac6c3b0163cad2474 100644
--- a/front_end/entrypoints/main/MainImpl.ts
+++ b/front_end/entrypoints/main/MainImpl.ts
@@ -753,6 +753,8 @@ export class MainImpl {
 globalThis.Main = globalThis.Main || {};
 // @ts-ignore Exported for Tests.js
 globalThis.Main.Main = MainImpl;
+// @ts-ignore Exported for Electron
+globalThis.EUI = UI || {};
 
 export class ZoomActionDelegate implements UI.ActionRegistration.ActionDelegate {
   handleAction(_context: UI.Context.Context, actionId: string): boolean {
